cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -119,8 +186,168 @@ pub fn maybe_make_service_config_file(
     Ok(service_config_path)
 }
 
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+struct ProcessState {
+    name: String,
+    namespace: String,
+    status: String,
+    system_time: String,
+    age: u64,
+    is_ready: String,
+    restarts: u64,
+    exit_code: i32,
+    pid: u64,
+    #[serde(rename = "IsRunning")]
+    is_running: bool,
+}
+
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+pub struct ProcessStates(Vec<ProcessState>);
+
+impl ProcessStates {
+    pub fn read(socket: impl AsRef<Path>) -> Result<ProcessStates, ServiceError> {
```

Comment:
suggestion nb: add a short comment what this does

```suggestion
	/// Read process states for an instance of process composed
	/// identified by the `socket`
    pub fn read(socket: impl AsRef<Path>) -> Result<ProcessStates, ServiceError> {
```

Reply:
Fixed